nginx/templates/_gzip.jinja

```diff
@@ -0,0 +1,29 @@
+{%- if gzip is defined and gzip is mapping %}
+{% from "nginx/lib.sls" import configurable %}
+  # Start configure gzip for vhost
+  gzip {{ 'on' if gzip.enable|default(true) else 'off' }};
```

Comment:
So, essentially you _could_ also do
```yaml
vhost.com:
   gzip:
     options:
        comp_level: 4
```

which doesn't require gzip to be explicitly enabled (`enabled: True`).